includes/class-sensei-enrolment-calculation-scheduler.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * File containing the class Sensei_Enrolment_Calculation_Scheduler.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * The Sensei_Enrolment_Calculation_Scheduler is responsible for running jobs of user enrolment calculations. It is set
+ * up to run once after a Sensei version upgrade or when Sensei_Course_Enrolment_Manager::get_site_salt() is updated.
+ */
+class Sensei_Enrolment_Calculation_Scheduler {
+	const CALCULATION_VERSION_OPTION_NAME = 'sensei-scheduler-calculation-version';
+
+	/**
+	 * Number of seconds between each job run.
+	 *
+	 * @var integer
+	 */
+	private $interval;
+
+	/**
+	 * Number of users for each job run.
+	 *
+	 * @var integer
+	 */
+	private $batch_size;
+
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+
+	/**
+	 * Sensei_Enrolment_Calculation_Scheduler constructor.
+	 *
+	 * @param integer $interval The scheduler's interval.
+	 * @param integer $batch_size The scheduler's batch size.
+	 */
+	private function __construct( $interval, $batch_size ) {
+		$this->interval   = $interval;
+		$this->batch_size = $batch_size;
+
+		// phpcs:ignore WordPress.WP.CronInterval.ChangeDetected -- Job size small.
+		add_filter( 'cron_schedules', [ $this, 'add_schedule' ] );
+		add_action( 'sensei_calculate_enrolments', [ $this, 'calculate_enrolments' ], 10, 0 );
+		add_action( 'init', [ $this, 'start' ], 101 );
+	}
+
+	/**
+	 * Fetches the instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self( 60, 2 );
```

Comment:
Was going from `20` to `2` intentional here?

Reply:
Nope :) I'll remove interval entirely and switch to 20 again in my next commit.

Reply:
Fixed in 4c803c3a9199b0bd13f991fbbf246074d61d7bd3